src/geode/io/model/vtm_brep_output.cpp

```diff
@@ -82,29 +82,35 @@ namespace
 
                 tasks[counter++] = async::spawn( [&block, this] {
                     const auto& mesh = block.mesh();
-                    const auto file = absl::StrCat( files_directory(),
-                        "/Block_", block.id().string(), ".vtu" );
                     if( const auto* tetra =
                             dynamic_cast< const geode::TetrahedralSolid3D* >(
                                 &mesh ) )
                     {
-                        save_tetrahedral_solid( *tetra, file );
+                        const auto file = absl::StrCat( files_directory(),
+                            "/Block_", block.id().string(), ".vtu" );
```

Comment:
I would have put that part before all the ifs, without the extension, and done geode::save_...(*tetra, absl::StrCat(file, ".vti") here).